solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrClientBuilderBase.java

```diff
@@ -38,7 +38,8 @@ public abstract class HttpSolrClientBuilderBase<
   protected Set<String> urlParamNames;
   protected Integer maxConnectionsPerHost;
   protected ExecutorService executor;
-  protected boolean useHttp1_1 = Boolean.getBoolean("solr.http1");
+  protected final boolean defaultUseHttp1_1 = Boolean.getBoolean("solr.http1");
+  protected Boolean providedUseHttp1_1;
```

Comment:
personally, I'm not a fan of "provided".  The vast majority of fields here are in-effect "provided", and they don't need this name prefix.

Reply:
I chose provided for this one, so that people in the future maybe think twice before using it instead of the `shouldUseHttp1_1()` method to check whether or not to use http1. Not a huge concern, and I can change it.